### DISCUSSION TITLE
Migrate to CodeMirror 6.0

### DISCUSSION THREAD

**Discussion by ghost** [uBlockOrigin/uBlock-issues#2146] [Ideas]
A stable release of CodeMirror 6 is now available: https://discuss.codemirror.net/t/codemirror-6-0-has-been-released/4498

Blog post: https://marijnhaverbeke.nl/blog/codemirror-6.html
Migration guide: https://codemirror.net/docs/migration/

---

Given that the new version focuses on touchscreen support, the migration could mean solving terrible editing experiences on mobile devices such as:
- https://github.com/uBlockOrigin/uBlock-issues/issues/1716
- https://github.com/mozilla-mobile/fenix/issues/20534 (Closed as an upstream issue)

The demo of CodeMirror is here. I have noticed that the issues I referred to above seems to have disappeared in v6.
v5: https://codemirror.net/5/
v6: https://codemirror.net/try/

**Comment by krystian3w** [comment]
The team organisation made repo "codemirror6"?

**Comment by krystian3w** [comment]
https://github.com/uBlockOrigin/uBlock-issues/discussions/2430#discussioncomment-4509074

**Comment by krystian3w** [comment]
https://github.com/gorhill/uBlock/commit/a60a5c277227415b3a39b484d60f65220cc207e2

**Reply by gorhill** [reply]
Why do you post these comments?

In case it's a way to remind me about the topic here, I am aware of CodeMirror 6, and I looked into, and it's more work than expected (this requires building locally, no more source files to just pick as needed) and for now I rather not deal with this.
